### PR TITLE
More porting from rep

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: vault/audit.go [hunk 1]
```diff
@@ -69,12 +72,15 @@ func (c *Core) enableAudit(entry *MountEntry) error {
 	}
 
 	// Generate a new UUID and view
-	entryUUID, err := uuid.GenerateUUID()
-	if err != nil {
-		return err
+	if entry.UUID == "" {
```
Comment (briankassouf) [inline vault/audit.go]: Up to you, I'm not sure where this value would come from but `_, err := uuid.ParseUUID()`  could be used to ensure it's a correct uuid

File: vault/audit.go [hunk 2]
```diff
@@ -155,6 +174,15 @@ func (c *Core) loadAudits() error {
 		}
 		c.audit = auditTable
 	}
+	if rawLocal != nil {
+		if err := jsonutil.DecodeJSON(rawLocal.Value, localAuditTable); err != nil {
+			c.logger.Error("core: failed to decode local audit table", "error", err)
+			return errLoadAuditFailed
+		}
+		for _, entry := range localAuditTable.Entries {
+			c.audit.Entries = append(c.audit.Entries, entry)
```
Comment (briankassouf) [inline vault/audit.go]: You could remove the for loop and do:
```go
c.audit.Entries = append(c.audit.Entries, localAuditTable.Entries...)
```

File: vault/audit.go (hunk 1 above)
Comment (jefferai) [inline vault/audit.go]: it comes from tests and may not always be valid, so happy to leave it alone.

File: vault/audit.go (hunk 2 above)
Comment (jefferai) [inline vault/audit.go]: Very true.

File: vault/cluster.go [hunk 3]
```diff
@@ -339,45 +339,67 @@ func (c *Core) stopClusterListener() {
...
 	}
 
 	// Prevent data races with the TLS parameters
 	c.clusterParamsLock.Lock()
 	defer c.clusterParamsLock.Unlock()
 
-	if c.localClusterCert == nil || len(c.localClusterCert) == 0 {
-		return nil, fmt.Errorf("cluster certificate is nil")
-	}
+	forwarding := c.localClusterCert != nil && len(c.localClusterCert) > 0
```
Comment (briankassouf) [inline vault/cluster.go]: Why don't we want to end early here if localClusterCert isn't set?

Comment (jefferai) [inline vault/cluster.go]: Basically in the enterprise case we may need to continue anyways so we just make the rest of the function do the right thing so the control flow can stay the same.

Comment (jefferai) [inline vault/cluster.go]: Of course assuming the rest does the right thing :-D

**Pull request merged**